lib/matplotlib/ticker.py

```diff
@@ -853,7 +853,7 @@ def set_locs(self, locs):
             vmax = math.log(vmax) / math.log(b)
             numdec = abs(vmax - vmin)
 
-        if numdec > 3:
+        if numdec > 1:
```

Comment:
My 2 cents is that we are trying to be too smart here and it has unintended consequences for anyone trying to use this formatter. At most, this sublabel filtering should be disabled by default (and possibly activated internally).

Also, the sublabel should probably (if kept) be private.